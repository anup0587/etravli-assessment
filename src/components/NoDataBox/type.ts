export interface NoImageProps {
  image: string;
  text: string;
}
