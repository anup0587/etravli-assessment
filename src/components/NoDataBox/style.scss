.no_image_box{
    display: flex;
    flex-direction: column;
    gap: 10px;
}