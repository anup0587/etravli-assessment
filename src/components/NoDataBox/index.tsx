import { Box, Typography } from "@mui/material";
import React from "react";
import "./style.scss"
import { NoImageProps } from "./type";

const NoDataBox = ({ image, text }: NoImageProps) => (
  <Box className="no_image_box">
    <img src={image} />
    <Typography variant="h5" gutterBottom>
      {text}
    </Typography>
  </Box>
);
export default NoDataBox;
