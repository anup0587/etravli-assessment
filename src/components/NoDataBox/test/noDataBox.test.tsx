import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import NoDataBox from "..";

describe("snapShot", () => {
  it("component snapshot", () => {
    const { container } = render(
      <NoDataBox image="/example/noData.png" text="no image" />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render an image and text", () => {
    const props = { image: "test-image.jpg", text: "No Data Found" };
    render(<NoDataBox {...props} />);
    const image = screen.getByRole("img");
    const text = screen.getByText(props.text);
    expect(image).toBeInTheDocument();
    expect(text).toBeInTheDocument();
  });
});
