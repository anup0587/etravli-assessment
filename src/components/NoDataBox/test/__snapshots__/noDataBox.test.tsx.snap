// Vitest Snapshot v1

exports[`snapShot > component snapshot 1`] = `
<div>
  <div
    class="no_image_box MuiBox-root css-0"
  >
    <img
      src="/example/noData.png"
    />
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
    >
      no image
    </h5>
  </div>
</div>
`;
